helpers/frontpanel.cpp

```diff
@@ -624,18 +627,21 @@ namespace WPEFramework
             LOGWARN("startBlinkTimer numberOfBlinkRepeats: %d m_blinkList.length : %d", numberOfBlinkRepeats, m_blinkList.size());
             stopBlinkTimer();
             m_numberOfBlinks = 0;
+            m_isBlinking = true;
             m_maxNumberOfBlinkRepeats = numberOfBlinkRepeats;
             m_currentBlinkListIndex = 0;
             if (m_blinkList.size() > 0)
             {
                 FrontPanelBlinkInfo blinkInfo = m_blinkList.at(0);
                 setBlinkLed(blinkInfo);
-                blinkTimer.Schedule(Core::Time::Now().Add(blinkInfo.durationInMs), m_blinkTimer);
+                if (m_isBlinking)
```

Comment:
Checking how will this work if we setBlink on different led indicators (power, data, record). Should we have different blink timers for each of these leds?

Reply:
Hi Anand,
Having multiple blink timers might fix issues while handling multiple LEDs, but it will still have this race condition between Revoke and Schedule functions. This fix avoids the overrides the race condition.

Reply:
ok sounds good. Will take this change and can look at adding multiple timers to handle other LEDs.